### PR TITLE
DBClient integration tests for MySQL, MadiaDB, PostgreSQL and MS SQL

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Added DBClient IT for more databases and JDBC drivers.

To run the tests, type:
```
cd <helidon_src>/tests/integration/dbclient
mvn clean install -D<db> -Pdocker -pl common,jdbc
```
Where
- `<helidon_src`> is your Helidon source tree directory
- and `<db>` is value from:
  - mysql
  - mariadb
  - pgsql
  - mssql

### CODE REVIEW

**Comment by Tomas-Kraus** [comment]
@romain-grecourt please check maven things and whether / how we can run this in our CICD environment
Theoretically every DB profile may run as parallel task.

**Comment by Tomas-Kraus** [comment]
@tomas-langer I did some changes in DbClientHealthCheck API. This needs to be checked.

File: dbclient/dbclient/src/main/java/io/helidon/dbclient/DbClient.java [hunk 1]
```diff
@@ -57,8 +57,14 @@
      */
     <U, T extends Subscribable<U>> T execute(Function<DbExecute, T> executor);
 
+    /**
+     * Name of the named statement used in database ping.
```
Comment (tomas-langer) [inline dbclient/dbclient/src/main/java/io/helidon/dbclient/DbClient.java:61]: used in database health checks

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java [hunk 2]
```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
...
-            dbClient.ping().await(10, TimeUnit.SECONDS);
+            execPing().await(timeoutSeconds, TimeUnit.SECONDS);
             builder.up();
         } catch (Throwable e) {
             builder.down();
             builder.withData("ErrorMessage", e.getMessage());
             builder.withData("ErrorClass", e.getClass().getName());
-            e.printStackTrace();
+            // Do not process logging arguments until it makes sense.
+            if (LOGGER.isLoggable(Level.FINER)) {
```
Comment (tomas-langer) [inline dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:94]: Please do not do this, use 
`Logger.FINER(() -> String.format("...")` or equivalent method with a lambda - it achieves the same thing without additional if statement in the code

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java [hunk 3]
```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
...
+    }
+
+    /**
+     * Database health check which calls default DBClient's {@code ping} method.
+     */
+    private static final class DbClientHealthCheckAsPing extends DbClientHealthCheck {
+
+        private DbClientHealthCheckAsPing(Builder builder) {
+            super(builder);
+            LOGGER.log(Level.FINEST, "Created an instance of DbClientHealthCheckAsPing");
```
Comment (tomas-langer) [inline dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:114]: `LOGGER.finest` instead

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java [hunk 4]
```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
...
+        }
+
+    }
+
+    /**
+     * Database health check which calls DBClient's {@code namedDml} method.
+     */
+    private static final class DbClientHealthCheckAsNamedDml extends DbClientHealthCheck {
+
+        /** Name of the statement. */
```
Comment (tomas-langer) [inline dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:129]: Please do not put Javadoc on private fields, use either single line or multi line comment instead.

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java [hunk 5]
```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
...
     public static final class Builder implements io.helidon.common.Builder<DbClientHealthCheck> {
+
+        /** Helidon database client. */
         private final DbClient database;
+        /** Health check name. */
         private String name;
+        /** Health check timeout in seconds (to wait for statement execution response). */
+        private int timeoutSeconds;
+
+        // Those two boolean variables define 4 ways of query execution:
```
Comment (tomas-langer) [inline dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:225]: Maybe a multiline comment would be a better fit?

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java [hunk 6]
```diff
@@ -104,6 +283,64 @@ public Builder name(String name) {
...
+        }
+
+        /**
+         * Set custom timeout to wait for statement execution response.
+         * Default value is {@code 10} seconds.
+         *
+         * @param seconds number of seconds to wait for statement execution response
+         * @return updated builder instance
+         */
+        public Builder timeout(int seconds) {
```
Comment (tomas-langer) [inline dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:339]: When configuring timeout in code, please use 
`timeout(long duration, TimeUnit timeUnit)`
or you need to name the method to be clear (e.g. `timeoutSeconds`) - in this case I would prefer the first one
If there is support for it in config, the configurable value must be `timeout-seconds` or `timeout-millis` and the code that converts it calls the method with TimeUnit)

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java [hunk 7]
```diff
@@ -209,7 +209,7 @@ public T apply(Throwable t) {
 
     @Override
     public Single<Void> ping() {
```
Comment (tomas-langer) [inline dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java:211]: This method should probably be deprecated, as the health check may be either DML or Query.

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: dbclient/dbclient/src/main/java/io/helidon/dbclient/DbClient.java (hunk 1 above)
Comment (Tomas-Kraus) [reply dbclient/dbclient/src/main/java/io/helidon/dbclient/DbClient.java:61]: fixed

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java (hunk 4 above)
Comment (Tomas-Kraus) [reply dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:129]: Changed all those somments to `/* ... */`

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java (hunk 3 above)
Comment (Tomas-Kraus) [reply dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:114]: fixed

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java (hunk 5 above)
Comment (Tomas-Kraus) [reply dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:225]: timeoutSeconds was replaced with timeoutDuration/timeoutUnit and comments updated.

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java (hunk 6 above)
Comment (Tomas-Kraus) [reply dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:339]: changed to `public Builder timeout(long duration, TimeUnit timeUnit)`.

File: dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java (hunk 2 above)
Comment (Tomas-Kraus) [reply dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java:94]: fixed

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java (hunk 7 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java:211]: Yes, this ping looks like some workaround or shortcut. User should use health check buillder/API to call proper statement.

**Review by tomas-langer — DISMISSED** [review]
LGTM

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**